gax-java/gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -258,6 +260,11 @@ public boolean needsEndpoint() {
     return endpoint == null;
   }
 
+  @Override
+  public boolean needsMtlsEndpoint() {
+    return mtlsEndpoint == null;
```

Comment:
If we want, I think we could make this:
```
return (mtlsEndpoint == null) && (useS2A == true)
```

That way, we can be sure we are only setting the `mtlsEndpoint` when we know:
- it is non-null
- it is non-empty
- it is needed by S2A

Note that the first two are checked in `shouldUseS2A` in EndpointContext.

Reply:
I think what you have is fine.